agent/agent.go

```diff
@@ -1188,6 +1198,9 @@ func (a *Agent) JoinLAN(addrs []string) (n int, err error) {
 	a.logger.Printf("[INFO] agent: (LAN) joining: %v", addrs)
 	n, err = a.delegate.JoinLAN(addrs)
 	a.logger.Printf("[INFO] agent: (LAN) joined: %d Err: %v", n, err)
+	if err == nil && a.joinLANNotifier != nil {
```

Comment:
It won't fire the event if you script your join from some outside thing that does `consul join` or uses the `/v1/agent` APIs, for example. I'd be ok leaving this as-is as long as we document that this event only comes out if you provide a `join` configuration to Consul when you start it, since this is a lot simpler to understand like this vs. the hook.